### PR TITLE
Rename suicide to selfdestruct

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: libethereum/Executive.cpp [hunk 1]
```diff
@@ -561,7 +561,8 @@ bool Executive::finalize()
     if (m_ext)
     {
         // Accumulate refunds for suicides.
-        m_ext->sub.refunds += m_ext->evmSchedule().suicideRefundGas * m_ext->sub.suicides.size();
+        m_ext->sub.refunds +=
+            m_ext->evmSchedule().suicideRefundGas * m_ext->sub.selfdestructs.size();
```
Comment (gumb0) [inline libethereum/Executive.cpp]: `selfdestructRefundGas`?

File: libevm/ExtVMFace.h [hunk 2]
```diff
@@ -239,8 +239,15 @@ class ExtVMFace
     /// Does the account exist?
     virtual bool exists(Address) { return false; }
 
-    /// Suicide the associated contract and give proceeds to the given address.
-    virtual void suicide(Address) { sub.suicides.insert(myAddress); }
+    /// Selfdestruct the associated contract and give proceeds to the given address.
+    ///
+    /// @param beneficiary  The address of the account which will receive ETH
+    ///                     from the selfdestructed account.
+    virtual void selfdestruct(Address beneficiary)
```
Comment (halfalicious) [inline libevm/ExtVMFace.h]: Should we leave this arg unnamed? It would enable us to skip adding the (void)beneficiary...is there a reason to name it?

**Comment by halfalicious** [comment]
Should we also change the opcode names in aleth-interpreter (e.g. in libaleth-interpreter/VM.cpp)

**Comment by halfalicious** [comment]
Should we update the comments in Account.h?

File: libethereum/Executive.cpp [hunk 3]
```diff
@@ -561,7 +561,8 @@ bool Executive::finalize()
     if (m_ext)
     {
         // Accumulate refunds for suicides.
```
Comment (halfalicious) [inline libethereum/Executive.cpp]: What about this comment?

**Comment by chfast** [comment]
I have run out of time on Friday. To be continued... I plan to also change Instructions and comments. The only "suicide" to be left are references to existing consensus tests.

**Pull request merged**